src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -17,22 +18,24 @@
 using Microsoft.CodeAnalysis.Shared.Utilities;
 using Microsoft.CodeAnalysis.Structure;
 using Microsoft.CodeAnalysis.SymbolMapping;
-using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.MetadataAsSource;
 
 [Export(typeof(IMetadataAsSourceFileService)), Shared]
 internal sealed class MetadataAsSourceFileService : IMetadataAsSourceFileService
 {
-    private const string MetadataAsSource = nameof(MetadataAsSource);
+    internal const string MetadataAsSource = nameof(MetadataAsSource);
 
     /// <summary>
     /// Set of providers that can be used to generate source for a symbol (for example, by decompiling, or by
     /// extracting it from a pdb).
     /// </summary>
     private readonly Lazy<ImmutableArray<Lazy<IMetadataAsSourceFileProvider, MetadataAsSourceFileProviderMetadata>>> _providers;
 
+    private readonly Lazy<FileSystemMetadataDocumentPersister> _fileSystemMetadataDocumentPersister = new(() => new FileSystemMetadataDocumentPersister());
```

Comment:
Debated between this and workspace services.  Ended up with this because we don't know at the time of workspace creation if we'll be using virtual files (we have to check the client capability).

Also thought about just passing through a flag instead of a type, but having a persister type consolidated things a bit nicer IMHO